test/test_sparse.cc

```diff
@@ -1,122 +1,214 @@
-#include "RandBLAS/config.h"
-#include "RandBLAS/sparse.hh"
-#include "RandBLAS/util.hh"
-#include "RandBLAS/test_util.hh"
-#include <cmath>
+#include <RandBLAS/dense.hh>
+#include <RandBLAS/sparse.hh>
+#include <RandBLAS/util.hh>
+#include <RandBLAS/test_util.hh>
 #include <gtest/gtest.h>
+#include <math.h>
 
 
+// TODO: ask Burlen what could cause the memory in S0.rows
+// to become deallocated / invalid after this function returns.
+//
+//      My guess is that C++ is returning a copy of S0
+//      but the copy is shallow, in that it copies the
+//      pointers S0.rows, S0.cols, S0.vals, but not the
+//      contents of those arrays.
+//
+//      If that guess is correct then I have another question:
+//      could this be avoided by using std::vector instead
+//      of plain arrays?
+//
+// Once I have an understanding of that, this function can be
+// removed (it's no longer used anywhere).
```

Comment:
@burlen I have a question for you. Some tests were failing erratically on my local machine even though they passed in CI. The cause of the failure was that the ``.rows`` field of a sketching operator returned from this function would become corrupted (nonsense values). I have some guesses above about what could have caused that. Can you comment on is my guesses are correct?

Reply:
I'll take a look. Which test fails?

I think this highlights some of the issues that smart pointers are designed to solve. For instance when a pointer copy occurs , who deletes the memory?

Reply:
I'd like to explore the shared pointers solution. For now, I'll remove this commented-out code so this can be merged.